src/app/ReadClient.h

```diff
@@ -237,6 +243,8 @@ class ReadClient : public Messaging::ExchangeDelegate
     ReadClient * GetNextClient() { return mpNext; }
     void SetNextClient(ReadClient * apClient) { mpNext = apClient; }
 
+    CHIP_ERROR SendSubscribeRequest(ReadPrepareParams && aReadPrepareParams);
```

Comment:
There has to be a fair bit of comments here, because the application has to know to a) allocate a `ReadPrepareParams` object that will have fields `mpEventPathParamsList` and  `mpAttributePathParamsList ` with lifetimes as long as the ReadClient itself and b) free' those up later in the call to `OnDeallocatePaths`.

Reply:
updated, thanks